### DISCUSSION TITLE
tauron_energy_bank ...jak to interpretowac ?

### DISCUSSION THREAD

**Discussion by mmg1t** [PiotrMachowski/Home-Assistant-custom-components-Tauron-AMIplus#194] [Q&A]
AMIplus zbiera dane do zakladki Energia - zdaje sie prawidłowo, jednak analiza ile zostało do odbioru ...jest mocno problematyczna.
Zgodnie z HowTo na stronie glownej projektu dodalem wpisy, i wykres juzyskałem taki jak ponizej ...
1. co oznaczają ujemne wartości ?
2. co tak na prawde zmienia "initial energy bank" ? ..mialem ustawione na jakas wartosc 200...co to znaczy?
3. jak odczytac ile moge odebrac energii ?

![tauron_energy_bank](https://github.com/user-attachments/assets/a4128e26-8d03-45cc-a348-9540bde786ff)

**Comment by PiotrMachowski** [comment]
Na początku zaznaczę, że nie jestem prosumentem, więc informacje na temat działania banku pochodzą od innych użytkowników integracji. Moje odpowiedzi na pytania są bardziej od strony implementacyjnej, niż jak to faktycznie jest liczone po stronie Taurona, ponieważ nie jestem do końca obeznany w aktualnych podejściach z ich strony.

> 1. co oznaczają ujemne wartości ?

Zapewne pusty bank

> 2. co tak na prawde zmienia "initial energy bank" ? ..mialem ustawione na jakas wartosc 200...co to znaczy?

Powinieneś tutaj ustawić wartość jaką miałeś w banku na począktu okresu rozliczeniowego (dla przykładu z README - na począku roku)

> 3. jak odczytac ile moge odebrac energii ?

Jeśli wszystko działa w porządku, to powinna ta wartość być zwracana przez encję "Tauron energy bank", ale najlepiej by było porównać ją z wartościami z eLicznika i jeśli się będą zgadzać, to powinno oznaczać, że wartość obliczana jest prawidłowo

**Comment by mmg1t** [comment]
tauron to skomplikowal strasznie :( ... z tego co rozumiem trzeba zobaczyc faktury i rozliczyc od okresu rozliczeniowego ... dzien po dniu... bo wg elicznika rozliczenie zbiorcze za pol roku nie jest poprawne ...
np.jesli 2giego dnia okresu rozliczeniowego pobierzemy megawat_h, ale w nadchodzacym polroczu odrobimy nawet 1,5megawatoh ...
wariant1.elicznikkowy - pokaze ze jestesmy naplusie
wariant2-prawdziwy: musmy zaplacic za ten pierwszy megawat a potem ew. zrobic sobie budget na kolejny okres z tego 1,5megawata...

Czy da sie jakos "wstecz" ustawic ten "inital energy bank" ? ...proboje odnalezc gdzie to jest w bazie i jaka tam jest logika ..ale nie idzie poki co ..

**Reply by PiotrMachowski** [reply]
Co rozumiesz żeby ustawić wstecz? Żeby historia się przeliczyła? 

BTW, odpowiadaj proszę w wątku

**Reply by mmg1t** [reply]
tak proboje to ogarnąć by zaczęło działac...bo rozliczenie mam bodaj w marcu(kwietniu?) ... i na tamten dzien chyba należało wpisać jakaś wartość po przeliczeniach z faktury... 
teraz jest straszne siano na tym wykresie , w sumie nic nie mowi, poza tym ze od lutego bardziej rosnie niż maleje (delta jest dodatnia)... ale ile liczbowo ? hmm....

PS. ok, racja z tym odpowiadaniem wątkowo, czytelnie jest, sorki i dzieki!

**Reply by lukaszzyla** [reply]
Cześć, czy komuś udało się w jakiś sposób to ogarnąć?
Mój wykres wygląda tak:
<img width="1567" height="505" alt="image" src="https://github.com/user-attachments/assets/4f88307c-edf3-4624-83df-1becc56afdf4" />
I nie ma wiele wspólnego z rzeczywistością.
Najlepszy rozwiązaniem, ale nie wiem jak to zrobić, byłaby mozliwość ustawienia input_number.initial_energy_bank na dany konkretny dzień w roku (ja wybrałbym pewnie fakturę z Lutego lub Marca, ponieważ po sezonie zimowym, przynajmniej u mnie - jestem wyczyszczony do zera).
Wtedy bierzemy taką fakturę za okres do końca Lutego, ustawiamy initial energy bank (jeśli coś nam zostało),  "zerujemy" tauron energy bank na ten dzień (w moim przypadku to by wsytarczyło) i na kolejne 12 miesięcy mielibyśmy w miarę sensowny pogląd na to co mamy w banku.
Czy komuś udało się to osiągnąć?
Czy istnieje jakiś sposób, żeby ograniczyć wartość energy bank wyłącznie do dodatnich wartości?
ujemny energy bank oznacza, że musieliśmy zapłacić za prąd więc chyba nie ma specjalnego sensu, a zaburza interesujące nas dane.
Dzięki!
